docs: update the PR checklist (#1314)
elastic/elastic-charts: .github/pull_request_template.md

### Checklist

<!-- Delete any items that are not applicable to this PR. -->
- [ ] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [ ] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)

> - [ ] The `:theme` label has been added and the @elastic/eui-design team has been pinged when there are `Theme` API changes

Reviewer: Using our handle in here is now actually going to ping us on every PR 😬

Reviewer 2: Oops 🤦🏼 , fixed in https://github.com/elastic/elastic-charts/commit/b4bb0641e5902e4b40863bd7eb5c2b66109e3617

### Checklist

<!-- Delete any items that are not applicable to this PR. -->
- [ ] The code changes are specific to a chart type, the chart type label is attached (e.g. `:xy`, `:partition`)
- [ ] The code changes are specific to one or more features, one or more feature labels are attached (e.g. `:interactions`, `:axis`)
- [ ] `Theme` API change is introduced, the `:theme` label is attached and the EUI team is pinged
- [ ] The closing issue/s is connected to a running GH project (if not, recheck with the team the priority)
- [ ] Any new public API export must be added to the `packages/charts/src/index.ts`
- [ ] Unit tests are added/updated to match the most common scenarios

> - [ ] The feature requires a specific explanation, documentation and/or a storybook story is added

Reviewer 2: ```suggestion
- [ ] The proper documentation and/or storybook story has been added or updated
```

> - [ ] Any added stories must import from `@elastic/charts` except for test data & storybook

Reviewer 2: We could prevent this with linting rules

> - [ ] The code is checked for cross-browser compatibility (Chrome, Firefox, Edge)

Reviewer 2: ```suggestion
- [ ] The code has been checked for cross-browser compatibility (Chrome, Firefox, Safari, Edge)
```

> - [ ] The visual changes are tested with every available theme (darkmode, lightmode, EUI related) 

Reviewer 2: ```suggestion
- [ ] Visual changes have been tested with all available themes including `dark`, `light`, `eui-dark` & `eui-light`
```
